Rover/mode_auto.cpp

```diff
@@ -82,21 +82,15 @@ void ModeAuto::update()
     switch (_submode) {
         case SubMode::WP:
         {
-            // check if we've reached the destination
-            if (!g2.wp_nav.reached_destination() || g2.wp_nav.is_fast_waypoint()) {
-                // update navigation controller
-                navigate_to_waypoint();
-            } else {
-                // we have reached the destination so stay here
-                if (rover.is_boat()) {
-                    if (!start_loiter()) {
-                        start_stop();
-                    }
-                } else {
-                    start_stop();
-                }
+            // boats loiter once the waypoint is reached
+            if (rover.is_boat() && g2.wp_nav.reached_destination() && !g2.wp_nav.is_fast_waypoint()) {
+                start_loiter();
```

Comment:
Were no longer checking the return here?

Reply:
OK, thanks, I'll keep it navigating if the loiter fails.  Previously it would "stop" but I think for boats it's best to keep navigating if it can't loiter.  There's actually almost no chance of the loiter failing.